src/test/scala/BankingAppTest.scala

```diff
@@ -0,0 +1,94 @@
+package com.scalaSparkSkillValidation
+
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import com.holdenkarau.spark.testing.SharedSparkContext
+import com.scalaSparkSkillValidation.utils.DataGenerator
+import org.apache.spark.sql.functions._
+
+/**
+ * Test suite for BankingApp functionalities.
+ * */
+class BankingAppTest extends AnyFlatSpec with Matchers with SharedSparkContext {
+
+  /**
+   * Set up common test data
+   *
+   * @param spark: SparkSession
+   * @return Unit: Doesn't return anything
+   * */
+  def setupTestData(spark: SparkSession): Unit = {
+    val (accounts, transactions, customers, accountHolders) = DataGenerator.generateSampleData(spark)
+    Seq(
+      (accounts, "accounts"),
+      (transactions, "transactions"),
+      (customers, "customers"),
+      (accountHolders, "accountHolders")
+    ).foreach { case (df, name) => df.createOrReplaceTempView(name) }
+    BankingApp.processBankingData(spark)
+  }
+
+  /**
+   * Test if BakingApp creates `joined_accounts` temporary view with the correct amount of rows.
+   * */
+  "BankingApp" should "create joined_accounts view correctly" in {
+    val spark = SparkSession.builder().config(sc.getConf).getOrCreate()
```

Comment:
Wouldn't be better to create one spark session for all the tests?

Reply:
Yes, I oversee the time and resources it takes for Spark to create and tear down sessions. I'll research how I can create and stop the SparkSession for all test cases.

Reply:
Test suite has been refactored an now it uses a single SparkSession created and terminated using _beforeAll_ and _afterAll_ traits